src/common/com/intellij/plugins/haxe/lang/psi/impl/HaxeReferenceImpl.java

```diff
@@ -681,7 +694,7 @@ private void addChildClassVariants(Set<HaxeComponentName> variants, HaxeClass ha
     }
   }
 
-  private static void addUsingVariants(Set<HaxeComponentName> variants, @Nullable HaxeClass ourClass, List<HaxeClass> classes) {
+  private static void addUsingVariants(Set<HaxeComponentName> variants, Set<HaxeComponentName> variantsWithExtension, @Nullable HaxeClass ourClass, List<HaxeClass> classes) {
```

Comment:
Extension methods doesn't have the first argument. But you can call it statically or as extension method. So it depends on how are you calling it. I needed to differentiate those entries in context. And that was a easy way of doing that without too much refactoring.